src/components/views/messages/TextualBody.js

```diff
@@ -162,6 +168,53 @@ module.exports = React.createClass({
         }
     },
 
+    pillifyLinks: function(nodes) {
+        const MemberAvatar = sdk.getComponent('avatars.MemberAvatar');
+        const RoomAvatar = sdk.getComponent('avatars.RoomAvatar');
+        for (let i = 0; i < nodes.length; i++) {
+            const node = nodes[i];
+            if (node.tagName === "A" && node.getAttribute("href")) {
+                const href = node.getAttribute("href");
+                // HtmlUtils transforms `matrix.to` links to local links, so match against
+                // user or room app links.
```

Comment:
Well, it looks like your code is right, although I'm failing to find where this happens. It probably should not, and leave the href as matrix.to but override the onclick like we do with permalinks and linkified users / rooms.